src/main/java/com/mathworks/ci/tools/MatlabInstaller.java

```diff
@@ -0,0 +1,258 @@
+import org.apache.commons.io.IOUtils;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.verb.POST;
+
+public class MatlabInstaller extends DownloadFromUrlInstaller {
+
+    private String version;
+    private String products;
+    private static String DEFAULT_PRODUCT = "MATLAB";
+
+    @DataBoundConstructor
+    public MatlabInstaller (String id) {
+        super (id);
+    }
+
+    public String getVersion () {
+        return this.version;
+    }
+
+    @DataBoundSetter
+    public void setVersion (String version) {
+        this.version = version;
+    }
+
+    public String getProducts () {
+        return this.products;
+    }
+
+    @DataBoundSetter
+    public void setProducts (String products) {
+        this.products = products;
+    }
+
+    @Override
+    public FilePath performInstallation (ToolInstallation tool, Node node, TaskListener log)
+        throws IOException, InterruptedException {
+        FilePath supportingExecutable = preferredLocation (tool, node);
+        String[] systemProperties = getSystemProperties(node);
+        FilePath expectedPath;
+        if(systemProperties[0].toLowerCase ().contains ("os x")) {
+            expectedPath = new FilePath (supportingExecutable, this.getVersion ()+".app");
+        } else {
+            expectedPath = new FilePath (supportingExecutable, this.getVersion ());
+        }
+        MatlabInstallable installable;
+        try {
+            installable = (MatlabInstallable) getInstallable (systemProperties);
+        } catch (Exception e) {
+            throw new InstallationFailedException (e.getMessage ());
+        }
+
+        getFreshCopyOfExecutables (installable, supportingExecutable);
+        makeDir (expectedPath);
+
+        int result = installUsingMpm (node, expectedPath, log);
+            if (result == 0) {
+                log.getLogger ().println (
+                    "MATLAB installation of version " + this.getVersion ()
+                        + " using mpm completed successfully!");
+            }
+        return expectedPath;
+    }
+
+    private int installUsingMpm (Node node, FilePath destination, TaskListener log)
+        throws IOException, InterruptedException {
+
+        Launcher matlabInstaller = node.createLauncher (log);
+        ProcStarter installerProc = matlabInstaller.launch ();
+
+        ArgumentListBuilder args = new ArgumentListBuilder ();
+        args.add (destination.getParent ().getRemote () + getNodeSpecificMPMExecutor (node));
+        args.add ("install");
+        appendReleaseToArguments (args, log);
+        args.add ("--destination=" + destination.getRemote ());
+        addMatlabProductsToArgs (args);
+        installerProc.pwd (destination).cmds (args).stdout (log);
+        int result;
+        try {
+            result = installerProc.join ();
+        } catch (Exception e) {
```

Comment:
It looks like we check for `latest-including-prerelease` in `appendReleaseToArguments`, but then we trim it and we don't ever add the flag to install the prerelease.

Reply:
Are you referring to the statement as i had it before here https://github.com/mathworks/jenkins-matlab-plugin/pull/364#discussion_r1801184560 .